javascript-carousel/main.js

```diff
@@ -0,0 +1,130 @@
+var $carousel = document.querySelector('.carousel');
+var $chevLeft = document.querySelector('.left');
+var $chevRight = document.querySelector('.right');
+var $image = document.querySelectorAll('.image');
+var $circle = document.querySelectorAll('.circle');
+
+var activeIndex = 0;
+
+$carousel.addEventListener('click', function () {
+  clearInterval(advanceIntervalId);
+
+  if (event.target === $chevLeft) {
+    if (activeIndex !== 0) {
+      activeIndex += -1;
+      hideImage();
+      revealImage(activeIndex);
+    }
+    activeIndex = $image.length - 1;
+    hideImage();
+    revealImage(activeIndex);
+  } else if (event.target === $chevRight) {
+    if (activeIndex !== $image.length - 1) {
+      activeIndex += 1;
+      hideImage();
+      revealImage(activeIndex);
+    }
+    activeIndex = 0;
+    hideImage();
+    revealImage(activeIndex);
+  } else if (event.target.getAttribute('class') === 'far fa-circle selector circle') {
+    for (var i = 0; i < $circle.length; i++) {
+      if ($circle[i].getAttribute('class') === 'fas fa-circle selector circle') {
+        $circle[i].setAttribute('class', 'far fa-circle selector circle');
+        $image[i].setAttribute('class', 'image hidden');
+      }
+    }
+    for (var j = 0; j < $circle.length; j++) {
+      if (event.target.getAttribute('id') === $circle[j].getAttribute('id')) {
+        $image[j].setAttribute('class', 'image');
+      }
+      event.target.setAttribute('class', 'fas fa-circle selector circle');
+    }
+  }
+
+  advanceIntervalId = setInterval(advance, 3000);
+});
+
+function hideImage() {
+  for (var i = 0; i < $image.length; i++) {
+    $image[i].setAttribute('class', 'image hidden');
+    $circle[i].setAttribute('class', 'far fa-circle selector circle');
+  }
+}
+
+function revealImage(index) {
+  $image[index].setAttribute('class', 'image');
+  $circle[index].setAttribute('class', 'fas fa-circle selector circle');
+}
```

Comment:
Try combining these in to one function named `revealImage` that takes an `index` that hides all except the images at that index as well as activates the correct circle.